File: PROG2-MIDTERM/MidtermLabWork1/determinant_solver.js
/**
 * Midterm Lab Work 1: 3x3 Matrix Determinant Solver
 * Student: Lord Clyde W. Pega
 * Section: BSCSIT 1203
 */

// ── Assigned Matrix ───────────────────────────────────────
const matrix = [
    [3, 2, 1],   
    [5, 8, 2],   
    [4, 6, 9]    
];

// ── Matrix Printer (Strict Alignment) ─────────────────────
function printMatrix(m) {
    // Top bracket: 6 spaces + corner + 16 spaces + corner
    console.log("      ┌                ┐"); 
    m.forEach(row => {
        // padStart(2) ensures single digits have a leading space.
        // The total string between │ │ is exactly 16 characters.
        const col1 = row[0].toString().padStart(2);
        const col2 = row[1].toString().padStart(2);
        const col3 = row[2].toString().padStart(2);
        console.log(`      │  ${col1}   ${col2}   ${col3}  │`);
    });
    // Bottom bracket
    console.log("      └                ┘");
}

function computeMinor(a, b, c, d) {
    return (a * d) - (b * c);
}

function solveDeterminant(m) {
    const line = "=".repeat(52);

    console.log(line);
    console.log("  3x3 MATRIX DETERMINANT SOLVER");
    console.log("  Student: Lord Clyde W. Pega");
    console.log("  Assigned Matrix:");
    console.log(line);
    printMatrix(m);
    console.log(line);

    // Step-by-step logic
    const m11 = computeMinor(m[1][1], m[1][2], m[2][1], m[2][2]);
    const m12 = computeMinor(m[1][0], m[1][2], m[2][0], m[2][2]);
    const m13 = computeMinor(m[1][0], m[1][1], m[2][0], m[2][1]);

    const c11 =  m[0][0] * m11;
    const c12 = -m[0][1] * m12;
    const c13 =  m[0][2] * m13;
    const det = c11 + c12 + c13;

    console.log(`\nExpanding along Row 1:`);
    console.log(`  Step 1 - Minor M11: det([${m[1][1]},${m[1][2]}],[${m[2][1]},${m[2][2]}]) = ${m11}`);
    console.log(`  Step 2 - Minor M12: det([${m[1][0]},${m[1][2]}],[${m[2][0]},${m[2][2]}]) = ${m12}`);
    console.log(`  Step 3 - Minor M13: det([${m[1][0]},${m[1][1]}],[${m[2][0]},${m[2][1]}]) = ${m13}`);

    console.log(`\n  Cofactor C11 = (+1) * ${m[0][0]} * ${m11} = ${c11}`);
    console.log(`  Cofactor C12 = (-1) * ${m[0][1]} * ${m12} = ${c12}`);
    console.log(`  Cofactor C13 = (+1) * ${m[0][2]} * ${m13} = ${c13}`);

    console.log(`\n  det(M) = ${c11} + (${c12}) + ${c13}`);
    console.log(line);
    console.log(`  ✓  DETERMINANT = ${det}`);
    console.log(line);
}

solveDeterminant(matrix);

File: PROG2-MIDTERM/MidtermLabWork1/DeterminantSolver.java
/**
 * =====================================================
 * Student Name    : Lordclyde Pega
 * Course          : BSCSIT 1203 Programming 2
 * Assignment      : Midterm Lab Work 1 — 3x3 Matrix Determinant Solver
 * School          : University of Perpetual Help System DALTA, Molino Campus
 * Date            : March 25, 2026
 * GitHub Repo     : https://github.com/pega-lordclyde/uphsd-cs-pega-lordclyde
 *
 * Description:
 * This program calculates the determinant of a 3x3 matrix using 
 * cofactor expansion along the first row. 
 * =====================================================
 */

public class DeterminantSolver {

    // ── SECTION 1: Matrix Declaration ───────────────────────────────────
    static int[][] matrix = {
        { 3, 2, 1 },   
        { 5, 8, 2 },   
        { 4, 6, 9 }    
    };

    // ── SECTION 2: 2×2 Determinant Helper ───────────────────────────────
    static int computeMinor(int a, int b, int c, int d) {
        return (a * d) - (b * c);
    }

    // ── SECTION 3: Matrix Printer (Strict Alignment) ────────────────────
    static void printMatrix(int[][] m) {
        // Top: 6 spaces + 1 char + 16 dashes + 1 char
        System.out.println("      ┌                ┐"); 
        for (int[] row : m) {
            // Each row uses exactly 16 spaces between the vertical bars
            // %3d (3) + space (1) + %3d (3) + space (1) + %3d (3) + padding = 16
            System.out.printf("      │  %2d   %2d   %2d  │%n", row[0], row[1], row[2]);
        }
        // Bottom: 6 spaces + 1 char + 16 dashes + 1 char
        System.out.println("      └                ┘");
    }

    // ── SECTION 4: Step-by-Step Determinant Solver ──────────────────────
    static void solveDeterminant(int[][] m) {
        String line = "=".repeat(52);

        System.out.println(line);
        System.out.println("  3x3 MATRIX DETERMINANT SOLVER");
        System.out.println("  Student: Lord Clyde W. Pega");
        System.out.println("  Assigned Matrix:");
        System.out.println(line);
        printMatrix(m);
        System.out.println(line);

        // Determinant Math
        int m11 = computeMinor(m[1][1], m[1][2], m[2][1], m[2][2]);
        int m12 = computeMinor(m[1][0], m[1][2], m[2][0], m[2][2]);
        int m13 = computeMinor(m[1][0], m[1][1], m[2][0], m[2][1]);

        int c11 =  m[0][0] * m11;
        int c12 = -m[0][1] * m12;
        int c13 =  m[0][2] * m13;
        int det = c11 + c12 + c13;

        System.out.printf("  Step 1 - Minor M11: %d%n", m11);
        System.out.printf("  Step 2 - Minor M12: %d%n", m12);
        System.out.printf("  Step 3 - Minor M13: %d%n", m13);
        System.out.printf("%n  ✓  DETERMINANT = %d%n", det);
        System.out.println(line);
    }

    public static void main(String[] args) {
        solveDeterminant(matrix);
    }
}